### PR TITLE
Added pistachio_pro

### PR DESCRIPTION
**Pull request by 7-rate**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pistachio_pro/matrix.c
Comment (zvecr) [inline keyboards/pistachio_pro/matrix.c:28]:
```suggestion

```

File: keyboards/pistachio_pro/lib/bme280.h
Comment (zvecr) [inline keyboards/pistachio_pro/lib/bme280.h:1]:
```suggestion
/* Copyright 2021 rate
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
#pragma once
```

File: keyboards/pistachio_pro/config.h
Comment (zvecr) [inline keyboards/pistachio_pro/config.h:41]:
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by 7-rate** [comment]
Thank you for the review.

File: keyboards/pistachio_pro/config.h
Comment (drashna) [inline keyboards/pistachio_pro/config.h:102]: The font in question appears to be 6x8 (5x7 with a 1px padding).  Setting it to 6 height is just cutting off the characters.

File: keyboards/pistachio_pro/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/pistachio_pro/keymaps/default/keymap.c:65]: snprintf will eat up ~1.5kb of firmware space.  If you're okay with that, that's fine.  But you can do this to output an int as a string:
```c
    uint8_t n = get_current_wpm();
    char wpm_counter[4];
    wpm_counter[3] = '\0';
    wpm_counter[2] = '0' + n % 10;
    wpm_counter[1] = ( n /= 10) % 10 ? '0' + (n) % 10 : (n / 10) % 10 ? '0' : ' ';
    wpm_counter[0] = n / 10 ? '0' + n / 10 : ' ';
```

File: keyboards/pistachio_pro/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/pistachio_pro/keymaps/default/keymap.c:76]: As above

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/pistachio_pro/config.h
Comment (7-rate) [reply keyboards/pistachio_pro/config.h:102]: No problem. The settings are as intended.

File: keyboards/pistachio_pro/keymaps/default/keymap.c
Comment (7-rate) [reply keyboards/pistachio_pro/keymaps/default/keymap.c:65]: 現状では追加機能の予定はないため、問題ありません。

**Review by drashna — APPROVED** [review]

**Comment by 7-rate** [comment]
Thank you for your review.
Would you please merge it?

**Comment by drashna** [comment]
> Thank you for your review.
> Would you please merge it?

We usually merge after getting two approvals on the PR

File: keyboards/pistachio_pro/readme.md
Comment (noroadsleft) [inline keyboards/pistachio_pro/readme.md:19]: Above this line, could you please add instructions on how to reset the keyboard into bootloader mode?

File: keyboards/pistachio_pro/rules.mk
Comment (noroadsleft) [inline keyboards/pistachio_pro/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/pistachio_pro/rules.mk
Comment (7-rate) [reply keyboards/pistachio_pro/rules.mk]: Solved with 844d24

File: keyboards/pistachio_pro/readme.md
Comment (7-rate) [reply keyboards/pistachio_pro/readme.md:21]: Solved with 844d24

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**
